src/sphinx/archetypes/java_app/index.rst

```diff
@@ -147,6 +147,67 @@ application arguments with ``--``. For example
     ./bin/my-app -Dconfig.resource=prod.conf -- -appParam1 -appParam2
 
 
+Multiple Applications
+=====================
+
+If you have multiple main classes then the ``JavaAppPackaging`` archetype provides you with two different ways of
+generating start scripts.
+
+1. A start script for each entry point. This is the default behaviour, when no ``mainClass in Compile`` is set
+2. One start script for the defined ``mainClass in Compile`` and forwarding scripts for all other main classes.
+
+Multiple start scripts
+----------------------
+
+No configuration is needed. SBT sets ``mainClass in Compile`` automatically to ``None`` if multiple main classes are
+discovered.
+
+**Example:**
+
+For two main classes ``com.example.FooMain`` and ``com.example.BarMain`` ``sbt stage`` will generate these scripts:
+
+.. code-block:: none
+
+   bin/
+     bar-main
+     bar-main.bat
+     foo-main
+     foo-main.bat
+
+
+Single start script with forwarders
+-----------------------------------
+
+Generates a single start script for the defined main class in ``mainClass in Compile`` and forwarding scripts for all
+other ``discoveredMainClasses in Compile``. The forwarder scripts call the defined start script and set the ``-main``
+parameter to the concrete main class.
+
+The start script name uses the ``executableScriptName`` setting for its name. The forwarder scripts use a simplified
+version of the class name.
+
+
+**Example:**
+
+The ``build.sbt`` has an explicit main class set.
+
+.. code-block:: scala
+
+    name := "my-project"
+    mainClass in Compile := Some("com.example.FooMain")
+
+For two main classes ``com.example.FooMain`` and ``com.example.BarMain`` ``sbt stage`` will generate these scripts:
+
+.. code-block:: none
+
+   bin/
+     bar-main
+     bar-main.bat
+     my-project
+     my-project.bat
+
+
+Now you can package your application as usual, but with multiple start scripts.
+
```

Comment:
Here I might go into a bit more detail on the differences between the main start script and what it means to have a "forwarder". Also, how do you disable the generation of forwarders?